tests/reg_tests/test_time_spectral_naca64A010.py

```diff
@@ -0,0 +1,176 @@
+# built-ins
+import unittest
+import os
+import copy
+import numpy
+
+# MACH testing class
+from adflow import ADFLOW
+from idwarp import USMesh
+
+# import the testing utilities that live a few directories up
+import reg_test_utils as utils
+
+from reg_default_options import adflowDefOpts
+
+from reg_aeroproblems import ap_naca64A010_time_spectral
+import reg_test_classes
+
+
+baseDir = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestSolve(reg_test_classes.RegTest):
+    """
+    Tests for time spectral case for an airfoil.
+    """
+
+    N_PROCS = 2
+    ref_file = "solve_euler_time_spectral_naca64A010.json"
+
+    def setUp(self):
+
+        # Introduce a transfer class for displacement transfer from struct to aero.
+        class Transfer:
+
+            # simplified transfer class
+            # converting csd displacement to cfd surface nodes
+
+            def __init__(self, alpha, xRot, aeroSolver):
+
+                # takes in displacement history
+
+                self.alpha = alpha
+                self.ntimeintervalsspectral = len(alpha)
+
+                self.aeroSolver = aeroSolver  # a shallow copy of CFD solver
+
+                self.xRot = xRot
+
+            def getUndeformedSurfaceNodes(self):
+
+                self.MDGroup = self.aeroSolver.allWallsGroup
+
+                self.cfdPts0 = self.aeroSolver.getSurfaceCoordinates(self.MDGroup, includeZipper=False)
+
+            def setDisplacements(self):
+
+                xRot = self.xRot
+                ntimeintervalsspectral = self.ntimeintervalsspectral
+                alpha = self.alpha  # notice a shallow copy introduced here; dont change the underlying obj!
+                cfdPoints_init = self.cfdPts0  # notice a shallow copy introduced here; dont change the underlying obj!
+
+                N_pts = cfdPoints_init.shape[0]
+
+                self.cfdPts = []
+
+                for sps in range(ntimeintervalsspectral):
+
+                    cfdPoints_deformed = numpy.zeros((N_pts, 3))
+
+                    ptch_loc = alpha[sps]
+
+                    cc = numpy.cos(ptch_loc)
+                    ss = numpy.sin(ptch_loc)
+
+                    for j in range(N_pts):
+
+                        cfdPoints_deformed[j, 0] = cc * (cfdPoints_init[j, 0] - xRot) + ss * cfdPoints_init[j, 1] + xRot
+                        cfdPoints_deformed[j, 1] = -ss * (cfdPoints_init[j, 0] - xRot) + cc * cfdPoints_init[j, 1]
+                        cfdPoints_deformed[j, 2] = cfdPoints_init[j, 2]
+
+                    self.cfdPts.append(cfdPoints_deformed)
+
+            def setVolumeMesh(self):
+
+                ntimeintervalsspectral = self.ntimeintervalsspectral
+
+                for sps in range(ntimeintervalsspectral):
+
+                    self.aeroSolver.mesh.setSurfaceCoordinates(self.cfdPts[sps])
```

Comment:
See my comment below, but similarly, why do we interact with aeroSolver.mesh here? Its probably fine but it would be cleaner if we either interact with the mesh object directly, or if the call needs to involve the aero solver, then we can add a method in the aero solver that does this with the mesh. 

I am being nitpicky here because people look at these tests as examples to use features, and anything we put here will be copy pasted whenever this feature is used down the line.

Reply:
Similar to the above, it is not intended to be run by itself for now.  it is meant to be run with pyaerostructure.